compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -276,6 +276,34 @@ impl<'a> Parser<'a> {
     ) -> PResult<'a, Box<Ty>> {
         let allow_qpath_recovery = recover_qpath == RecoverQPath::Yes;
         maybe_recover_from_interpolated_ty_qpath!(self, allow_qpath_recovery);
+
+        if (self.token.is_keyword(kw::Const) || self.token.is_keyword(kw::Mut))
```

Comment:
This probably should also be moved into a separate function and also moved way down into the actually `if else` chain below rather than being hit first on every call to `parse_ty_common`